src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IncomingMessage } from 'http'
+import * as request from 'request'
+import { getLogger, Logger } from '../logger'
+import { ResourceFetcher } from './resourcefetcher'
+
+export class HttpResourceFetcher implements ResourceFetcher {
+    private readonly logger: Logger = getLogger()
+
+    public constructor(private readonly url: string) {}
+
+    /**
+     * Returns the contents of the resource, or undefined if the resource could not be retrieved.
+     */
+    public async get(): Promise<string | undefined> {
+        try {
+            this.logger.verbose(`Loading resource from ${this.url}`)
+
+            return new Promise<string>((resolve, reject) => {
+                request(this.url, (error: any, response: IncomingMessage, body: any) => {
```

Comment:
How do we get proxy settings in from the IDE - is this something we've had to tackle before? Does VS Code have any IDE-wide proxy configuration that we should be honouring?

Reply:
This is a relocation of existing http getter code. I need to learn more about proxy support, because no part of the toolkit currently accounts for proxy handling.